packages/core/src/services/scales-cartesian.ts

```diff
@@ -754,7 +754,11 @@ export class CartesianScales extends Service {
 
 		const { thresholds } = axesOptions[domainAxisPosition];
 
-		if (!thresholds) {
+		// Check if thresholds exist & is not empty
+		if (
+			!Array.isArray(thresholds) ||
+			(Array.isArray(thresholds) && !thresholds.length)
+		) {
 			return null;
 		}
```

Comment:
This check is done in only 2 places - `getHighestDomainThreshold()` and `getHighestRangeThreshold()` functions